NGR Kali Andhi-class Destroyer

Apogee-dev opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![kali_andhi_ingame](https://github.com/user-attachments/assets/dff8226f-0cfd-4086-be04-34941df001ab)
![kali_andhi](https://github.com/user-attachments/assets/71eaa7d3-ac03-4721-b0c4-001e0d686acc)
![kali_andhi_areas](https://github.com/user-attachments/assets/16497551-4328-49ee-932d-247a80079168)


Adds the Kali Andhi-class, a new class of NGR destroyer that makes up the bulk of the NGR's young fleet. This class's activity in securing NGR's territorial claims has made them an unwelcome sight for the PGF especially. The Kali Andhi is a combat ship, comparable to the Crying Sun or the Colossus, featuring good armament and plenty of space for the crew to interact. Crew count is as follows:

- 1 NGR Captain
- 1 Lieutenant; Commands the Operatives, organizes and executes away missions
- 1 Ensign; Mans the bridge, manages communications, coordinates the support crew
- 3 Operatives; Soldiers, plain and simple. Can be armed with a Sidewinder, a Hydra DMR, or an Asp.
- 1 Wrecker; Organize and handle the storage bay, retrieve resources, and occasionally scout for the Operatives
- 1 Mechanic; Operate and maintain the power and life support systems
- 1 Medical Doctor; Maintain the crew's general health, organize and manage medical supplies, bag and tag corpses
- 2 Initiates; General-purpose muscle for cleaning and maintaining the ship.

Also adds a set of NGR Tan floor decals, NGR wall and desk flags, and tweaks a few of the NGR outfits.
![kali_salute](https://github.com/user-attachments/assets/98f17072-4b00-4c44-ab79-f74f744aa09c)

## Why It's Good For The Game

NGR needs a ship and the Hyena doesn't count.

## Changelog

:cl:
add: the NGR Kali Andhi-class destroyer
add: NGR floor decals and flags
balance: red space suit armor made equivalent to sec hardsuits
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Apogee-dev committed:
draft 1

Apogee-dev committed:
size adjustment

Apogee-dev committed:
config setup

Apogee-dev committed:
draft 2: the tanning

Apogee-dev committed:
tweaks

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
the detailening + syndie softsuit tweak

Apogee-dev committed:
decor done

Apogee-dev committed:
wallmounts and pipes

Apogee-dev committed:
final touches

Apogee-dev committed:
tweaks

Erikafox commented on _maps/configs/syndicate_ngr_kaliandhi.json:
```diff
@@ -0,0 +1,54 @@
+	],
+	"map_name": "Kali Andhi-class Destroyer",
+	"map_short_name": "Kali Andhi-class",
+	"description": "A post-ICW destroyer design commissioned by the New Gorlex Republic and outfitted for general patrol and reclamation duties. The Kali Andhi-class, despite its young age, has been instrumental in securing New Gorlex's modest territorial claims, making them a somewhat unwelcome sight for the PGF's frontier mission.",
```
You should add a fluff bit about what its named for

Apogee-dev replied:
It's just named after a windstorm (a pattern I want to repeat in other NGR ships to make the fact that it's a pattern... yknow, obvious)

tantspants commented:
Great ship. I'm glad NGR is getting a real combat vehicle

Apogee-dev committed:
pipe fix

Apogee-dev committed:
centered captain's intercoms

Apogee-dev committed:
more pipefix

meemofcourse commented on code/modules/clothing/spacesuits/syndi.dm:
```diff
@@ -13,8 +13,10 @@
 	w_class = WEIGHT_CLASS_NORMAL
-	allowed = list(/obj/item/gun, /obj/item/ammo_box, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/transforming/energy/sword/saber, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
-	armor = list("melee" = 30, "bullet" = 15, "laser" = 15, "energy" = 40, "bomb" = 20, "bio" = 100, "rad" = 30, "fire" = 75, "acid" = 75)
+	/obj/item/clothing/suit/space/syndicate/Initialize()
```
```suggestion
/obj/item/clothing/suit/space/syndicate/Initialize()
```

Apogee-dev committed:
not sure how that happened

Apogee-dev committed:
faction fix

Apogee-dev committed:
begone random magazine

Apogee-dev committed:
clutter

Apogee-dev committed:
combustion thruster guide

kaylexis commented:
its so peak..

Sadhorizon commented on code/modules/clothing/outfits/factions/syndicate.dm:
```diff
@@ -521,6 +516,20 @@
+	suit = /obj/item/clothing/suit/armor/ngr/lieutenant
+	id = /obj/item/card/id/syndicate_command/crew_id
+	shoes = /obj/item/clothing/shoes/combat
+	suit_store = /obj/item/gun/ballistic/automatic/pistol/ringneck
```
Should they *really* spawn in with a gun? Isn't "gear in closets" rule a thing?

Sadhorizon commented on code/modules/clothing/outfits/factions/syndicate.dm:
```diff
@@ -594,7 +603,8 @@
 	id = /obj/item/card/id/syndicate_command/crew_id
 	shoes = /obj/item/clothing/shoes/combat
-	suit_store = /obj/item/gun/ballistic/automatic/pistol/ringneck
+	suit_store = /obj/item/gun/ballistic/automatic/pistol/asp
```
same here ig

Apogee-dev replied:
The Ensign one was actually a mistake because I started by copying the HoS datum, but the HoS spawns with a gun because he already spawned with a gun before I touched this, and so does every other syndicate HoS. I could remove it and shuffle the Asp to his locker, I suppose, but I won't change the Hardliner one or the base one. that'll have to wait until imaginos maps the hardliner carina- I don't want to touch the Komodo in this PR.

Apogee-dev committed:
hos gun

Apogee-dev committed:
asp empty mags

Apogee-dev committed:
soap.

Sadhorizon commented on _maps/configs/syndicate_ngr_kaliandhi.json:
```diff
@@ -0,0 +1,54 @@
+	"map_path": "_maps/shuttles/syndicate/syndicate_ngr_kaliandhi.dmm",
+	"map_id": "syndicate_gorlex_kaliandhi",
+	"limit": 1,
+	"job_slots": {
```
you forgot to add the wrecker

Apogee-dev committed:
forgor wrecker slot

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
stamps

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
hope this isnt too much gun

Apogee-dev committed:
sirocco fix

1glitchycent commented:
https://github.com/shiptest-ss13/Shiptest/pull/414 holy crap voulgier hd 1080p

Apogee-dev committed:
bridge electro windows

Apogee-dev committed:
ship access

Apogee-dev committed:
forgor the restroom

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
all this for a flag?

Apogee-dev committed:
all THIS for a FLAG?!

Apogee-dev committed:
light switch

goober3 reviewed (dismissed):
### hi, i'm the maptainer and i will now be conducting my review

**concerning the storage bay:**
-the crates on the shelves in here don't spawn on the shelves. this feels like something of an oversight
-consider giving them a ration crate. the food supplies in the fridge feel quite barren.
-also, put a couple extra emergency O2 tanks in the closet here. for the spacesuits. 
-there is a missing floor tile under the door to this room.

**concerning the armory:**
-there is an extra syndicate mask in the equipment locker. cast it away.
-i understand the need for vent/scrubber symmetry but please don't route wires & pipes under the armory office windoor.
-also, change the layer of the windoor in the office so it layers under the shutters.
-consider swapping out the container for the ammunition locker for something else, and/or add some railings here. something about its placement feels awkward. (see image)
![image](https://github.com/user-attachments/assets/ab963c27-010e-44e8-9d03-af48c6a281f6)

**everything else**
-there is a missing wire under the bridge door. this is bad because that room needs power.
-the note in atmospherics might be better flavored as a pamphlet or a book.
-please put a cigarrette vendor somewhere on this ship. smokers will suffer.

that's all i can come up with for now. we'll see how it performs in testing.

Erikafox commented:
>crates on shelves

They don't currently support that but I'm sure a helper or something for it would be easy

Apogee-dev commented:
> Crate shelves

They don't support spawning with crates loaded. I'll move them to adjacent tiles on the cargo bay floor.

> Rations

There is in fact a rations crate- it's one of the two crates that spawns on the same tiles as the shelves, and it's full of random MREs and bean cans.

> Oxygen

I didn't include any emergency tanks with the suits themselves because of the dispenser almost next door in the entrance bay- I'll throw some tiny ones in the SSUs.

> Syndie mask

That's for the Lieutenant, because he needs access to a proper gas mask just like the Operatives but it shouldn't spawn in his personal locker. There's an extra SSU here for the same reason.

> Windoor layer

Why this isn't the default is beyond me, honestly.

> Ammo container

Yeah I tried a great many layouts for the armory to make this work, but no layout using shelves really felt right and the ammo takes up too much space to shelf (it should reasonably be locked up anyway). I also despise railings for any purpose other than guarding height differences/stairs with a burning passion. I think this is fine enough.

> Atmos note

This note is a predefined guide, and it's the exact same one used on the Crying Sun. I'd rather not change it in this PR if it would affect another ship.

> Everything else

I'll fix that in the next commit.

Apogee-dev committed:
review fixes

meemofcourse committed:
Merge remote-tracking branch 'upstream/master' into pr/3554

meemofcourse commented on code/modules/clothing/outfits/factions/syndicate.dm:
```diff
@@ -715,6 +723,7 @@
 	back = /obj/item/storage/backpack/duffelbag/syndie/med
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1, /obj/item/storage/firstaid/medical,)
```
```suggestion

```

meemofcourse commented on code/modules/clothing/outfits/factions/syndicate.dm:
```diff
@@ -727,6 +736,7 @@
 	back = /obj/item/storage/backpack/duffelbag/syndie/med
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1, /obj/item/storage/firstaid/medical,)
```
```suggestion

```

meemofcourse commented on code/modules/clothing/outfits/factions/syndicate.dm:
```diff
@@ -727,6 +736,7 @@
 	back = /obj/item/storage/backpack/duffelbag/syndie/med
```
since this is a shipside outfit back is overriden by backpack, duffelbag, courierbag and satchel

meemofcourse reviewed (dismissed):
![imagen](https://github.com/user-attachments/assets/8ab38155-d8c0-4885-9235-3eab04da5164)
![imagen](https://github.com/user-attachments/assets/7581e5e2-d97e-45fd-bcbb-c2333acef269)

the comms console and the id console still kinda suck. you might wanna give them a modcomp, and i cannot spot a cargo console anywhere, so maybe that too

![imagen](https://github.com/user-attachments/assets/e588a5f8-8b96-43e7-8eea-bd3ad16d9420)
a problem with having multiple intercoms in one tile is that using :i will speak through both of them

![imagen](https://github.com/user-attachments/assets/e12718a5-8b07-43fb-863b-7adc8706463b)
![imagen](https://github.com/user-attachments/assets/bf7f6116-45a8-4a7c-a4a2-aead5657ce09)

the way the first cabinet is pixelshifted makes it seem as though you could stand on its same tile like the second one, but it is not the case

![imagen](https://github.com/user-attachments/assets/9b7509ae-012a-48d3-b602-e8b65cd1f714)
i think the arrangement of the table behind the bed and next to the bookcase seems a bit strange, but it's not a big deal

![imagen](https://github.com/user-attachments/assets/783c39be-d597-4324-8e8b-210f09ffe9ae)
change this

![imagen](https://github.com/user-attachments/assets/9434aac1-914d-4aed-8986-47d3cd28cb14)
![imagen](https://github.com/user-attachments/assets/93f18346-0ff1-4488-9493-08d33a01cbda)

* ngr medics seem to spawn with an additional box of syndie internals, alongside a medkit (with another in their hand) and a knife
* engis spawn with a full toolbelt
* wreckers have a lot of mining garbage
* operatives start with a loaded stun baton and full armor

 i know this is not enforced on all outfits and factions but try to make sure none of them spawn with gear that can be put in lockers or their respective workplaces

![imagen](https://github.com/user-attachments/assets/1f6c2d7e-b3f1-4422-81ed-8b7bf6f07aea)
![imagen](https://github.com/user-attachments/assets/d3550b1d-756f-4110-8229-56722c9be996)

ensign IDs and closet say Bridge Officer. if this is intentional you may carry on

cool ship though. i think its the first where i've been left wondering where the rooms are

Apogee-dev commented:
>comms console

Actually a mapper error, that's supposed to be an outpost console.

>modcomp

Modcomps don't work with the connected computer console sprites- the ID console works _okay_ on its own and ensures they at least have that functionality available, but I wanted to spread the consoles around a bit on the bridge since there's so many on this ship.

>intercoms

I didn't want to, but I guess I can remove the NGR shortwave intercom here to have just wideband and standard.

> spawn gear

Some of these are actually standard - Wrecker's backpack contents are the same as every other miner job in the game (sans the voucher, which i'm removing), as are secoffs spawning with armor and engis starting with tools. Everything else I'll cull though.

>everything else

Good eye, I'll get it all fixed.

Apogee-dev committed:
review

Apogee-dev committed:
Merge branch 'ngr-cruiser' of https://github.com/Apogee-dev/Shiptest into ngr-cruiser

Apogee-dev committed:
knife.

Apogee-dev committed:
thruster fix

Apogee-dev committed:
turret edits

moved heavy turret at nose, downgraded most turrets to light, shuffled turret coverage

Apogee-dev committed:
windows

Apogee-dev committed:
handrails

Apogee-dev committed:
enforces NGR Navy alcohol & tobacco policy

Apogee-dev committed:
too many signs

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
guncases.

Apogee-dev commented:
if this thing gets merge conflicted one more time by a newer refactor PR i'm closing it. i'm so tired.

Apogee-dev committed:
guncases again

Apogee-dev committed:
janitor basin

Apogee-dev committed:
miner and engi balaclava

Apogee-dev committed:
engi gas mask

Apogee-dev committed:
miner belt

Apogee-dev committed:
megaphones

Apogee-dev committed:
fixes the viper

Apogee-dev committed:
sirocco desc grammar

Apogee-dev committed:
cell charger

Leon9621 commented:
Having just played a round, wanted to add in some comments:

The wrecker's locker contains some default station cargo tech equipment: Stuff like a headset, uniforms, armbands, etc. Might want to take a look at that.

The main hallway is nice, but it feels... desolate. You can't see into any room, neither through airlocks nor windows, you're just kind of Locked In.

Apogee-dev committed:
depopulates wrecker locker

Apogee-dev commented:
> Having just played a round, wanted to add in some comments:
> 
> The wrecker's locker contains some default station cargo tech equipment: Stuff like a headset, uniforms, armbands, etc. Might want to take a look at that.
> 
> The main hallway is nice, but it feels... desolate. You can't see into any room, neither through airlocks nor windows, you're just kind of Locked In.

Good catch on the wrecker locker, I forgot to turn populate off on it. As for the main hallway, that's intentional; there's portholes into medbay, canteen, and living quarters so you can see if someone's in there, but the hallway's meant to feel like the inside of a military ship, not a civilian one; the Kali Andhi's internal spaces are generally fairly closed-in as an intentional design trait.

Sadhorizon commented:
> if this thing gets merge conflicted one more time by a newer refactor PR i'm closing it. i'm so tired.

Oh well, it had a good run.

Sun-Soaked committed:
Merge branch 'master' into ngr-cruiser

Signed-off-by: Sun-Soaked <45698967+Sun-Soaked@users.noreply.github.com>

Sun-Soaked commented:
> > if this thing gets merge conflicted one more time by a newer refactor PR i'm closing it. i'm so tired.
> 
> Oh well, it had a good run.

but what if I fixed the conflicts

Sun-Soaked commented on code/modules/clothing/spacesuits/syndi.dm:
```diff
@@ -14,7 +14,7 @@
 	allowed = list(/obj/item/gun, /obj/item/ammo_box, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/energy/sword/saber, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
-	armor = list("melee" = 30, "bullet" = 15, "laser" = 15, "energy" = 40, "bomb" = 20, "bio" = 100, "rad" = 30, "fire" = 75, "acid" = 75)
+	armor = list("melee" = 30, "bullet" = 15, "laser" = 30, "energy" = 40, "bomb" = 20, "bio" = 100, "rad" = 30, "fire" = 75, "acid" = 75)
 
```
```suggestion

/obj/item/clothing/suit/space/syndicate/Initialize()
		. = ..()
		allowed = GLOB.security_hardsuit_allowed

```

Sun-Soaked replied:
accidentally removed during conflict fix

Sun-Soaked commented on code/modules/clothing/spacesuits/syndi.dm:
```diff
@@ -14,7 +14,7 @@
 	desc = "A space suit made of high-grade ballistic fabric with integrated armor plates. More compact than a normal space suit while almost matching powered hardsuits for protection. Almost."
 	w_class = WEIGHT_CLASS_NORMAL
 	allowed = list(/obj/item/gun, /obj/item/ammo_box, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/energy/sword/saber, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
```
```suggestion
```

Sun-Soaked replied:
I accidentally removed this during the conflict fix

Sun-Soaked committed:
Update code/modules/clothing/spacesuits/syndi.dm

Signed-off-by: Sun-Soaked <45698967+Sun-Soaked@users.noreply.github.com>

Sun-Soaked committed:
Update code/modules/clothing/spacesuits/syndi.dm

Signed-off-by: Sun-Soaked <45698967+Sun-Soaked@users.noreply.github.com>

Apogee-dev committed:
spacing fix

Apogee-dev committed:
Merge branch 'ngr-cruiser' of https://github.com/Apogee-dev/Shiptest into ngr-cruiser

Apogee-dev committed:
dorm canteen window

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
mask path fix

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser

Apogee-dev committed:
moves thruster blast doors

Apogee-dev committed:
fine. extra briefing room seats

Apogee-dev committed:
not fine. no extra briefing room seats

Apogee-dev commented:
actively shriveling. ill fix the merge conflict this evening.

generalthrax commented:
This ship has been tested a fair few times in game already. Its probably good to fullmerge (once merge conflict 73871128 is fixed) to prevent being locked in merge conflict hell.

Apogee-dev committed:
Merge remote-tracking branch 'upstream/master' into ngr-cruiser
